WordPressComStatsiOS/WordPressComStatsiOS/UI/WPStatsGraphViewController.m

```diff
@@ -208,6 +217,44 @@ - (void)setVisits:(StatsVisits *)visits forSummaryType:(StatsSummaryType)summary
 
 #pragma mark - Private methods
 
+- (CGFloat)calculateDateFontSize
+{
+    // step through each of the dates
+    // and calculate what font size each would actually be displayed at
+    // then return the smallest font size out of all of them
+    CGFloat smallestFontSize = [[WPStyleGuide axisLabelFont] pointSize];
+
+    for (StatsSummary *summary in self.statsData) {
+        NSInteger index = (NSInteger)[self.statsData indexOfObject:summary];
+        NSIndexPath *indexPath = [NSIndexPath indexPathForItem:index inSection:0];
+
+        WPStatsGraphBarCell *cell = [self collectionView:self.collectionView cellForItemAtIndexPath:indexPath];
+        CGFloat currentFontSize = [self fontSizeForString:cell.barName inRect:cell.frame];
+
+        if (index == 0) {
+            smallestFontSize = currentFontSize;
+        } else if (currentFontSize < smallestFontSize) {
+            smallestFontSize = currentFontSize;
+        }
+    }
+
+    return smallestFontSize;
+}
+
+- (CGFloat)fontSizeForString: (NSString *)string inRect: (CGRect)rect
```

Comment:
Since only the `rect.size` is ever used, would it be possible to update the signature?

```
- (CGFloat)fontSizeForString: (NSString *)string containerSize: (CGSize)size
```

Or something like that! naming isn't really important. But it'd be great to ditch the rect's origin, since it's not required here.